webapp/src/Controller/PublicController.php

```diff
@@ -116,6 +95,114 @@ public function scoreboardAction(Request $request): Response
+        }
+        $zip->addFromString('index.html', $contestPage);
+
+        $publicPath = realpath(sprintf('%s/public/', $projectDir));
+        foreach ($assetMatches[1] as $file) {
+            $zip->addFile($publicPath . '/' . $file, $file);
+        }
+
+        // Also copy in the webfonts
+        $webfontsPath = $publicPath . '/webfonts/';
+        foreach (glob($webfontsPath . '*') as $fontFile) {
+            $fontName = basename($fontFile);
+            $zip->addFile($fontFile, 'webfonts/' . $fontName);
+        }
+
+        $zip->close();
+
+        $zipFilename = 'contest.zip';
+
+        $response = new StreamedResponse();
+        $response->setCallback(function () use ($tempFilename) {
+            $fp = fopen($tempFilename, 'rb');
+            fpassthru($fp);
+            unlink($tempFilename);
+        });
+        $response->headers->set('Content-Type', 'application/zip');
+        $response->headers->set('Content-Disposition', 'attachment; filename="' . $zipFilename . '"');
+        $response->headers->set('Content-Length', filesize($tempFilename));
+        $response->headers->set('Content-Transfer-Encoding', 'binary');
+        $response->headers->set('Connection', 'Keep-Alive');
+        $response->headers->set('Accept-Ranges', 'bytes');
+
+        return $response;
+    }
+
+    /**
+     * Get the contest from the request, if any
+     */
+    protected function getContestFromRequest(Request $request): ?Contest
+    {
+        $contest = null;
+        // For static scoreboards, allow to pass a contest= param.
+        if ($contestId = $request->query->get('contest')) {
+            if ($contestId === 'auto') {
+                // Automatically detect the contest that is activated the latest.
+                $contest      = null;
+                $activateTime = null;
+                foreach ($this->dj->getCurrentContests(-1) as $possibleContest) {
+                    if (!$possibleContest->getPublic() || !$possibleContest->getEnabled()) {
+                        continue;
+                    }
+                    if ($activateTime === null || $activateTime < $possibleContest->getActivatetime()) {
+                        $activateTime = $possibleContest->getActivatetime();
+                        $contest      = $possibleContest;
+                    }
+                }
+            } else {
+                // Find the contest with the given ID.
+                $contest = null;
+                foreach ($this->dj->getCurrentContests(-1) as $possibleContest) {
+                    if ($possibleContest->getCid() == $contestId || $possibleContest->getExternalid() == $contestId) {
+                        $contest = $possibleContest;
+                        break;
+                    }
+                }
+
+                if (!$contest) {
+                    throw new NotFoundHttpException('Specified contest not found');
```

Comment:
```suggestion
                    throw new NotFoundHttpException('Specified contest not found.');
```

You also add it for the error in https://github.com/DOMjudge/domjudge/pull/1641/files#diff-cc71aa6a9b851ee37894e1943a11fd1d6676ae347fac6e088ec93325e4a90e1aR130

Reply:
Well no, I copied it, but it makes sense.